scripts/install-worker.sh

```diff
@@ -328,6 +359,64 @@ if [[ ! $KUBERNETES_VERSION =~ "1.19"* && ! $KUBERNETES_VERSION =~ "1.20"* && !
+### Cache Images ###############################################################
+################################################################################
+if [[ "$CACHE_CONTAINER_IMAGES" = "true" && "$BINARY_BUCKET_REGION" != "us-iso-east-1" && "$BINARY_BUCKET_REGION" != "us-isob-east-1" ]]; then
+    AWS_DOMAIN=$(curl -s http://169.254.169.254/latest/meta-data/services/domain)
+    ECR_URI=$(/etc/eks/get-ecr-uri.sh "${BINARY_BUCKET_REGION}" "${AWS_DOMAIN}")
+
+    PAUSE_CONTAINER="${ECR_URI}/eks/pause:${PAUSE_CONTAINER_VERSION}"
+    KUBE_PROXY_IMG="${ECR_URI}/eks/kube-proxy:v${KUBERNETES_VERSION}-${KUBE_PROXY_VERSION_SUFFIX}"
+    KUBE_PROXY_MINIMAL_IMG="${ECR_URI}/eks/kube-proxy:v${KUBERNETES_VERSION}-minimal-${KUBE_PROXY_VERSION_SUFFIX}"
+    cat /etc/eks/containerd/containerd-config.toml | sed s,SANDBOX_IMAGE,$PAUSE_CONTAINER,g | sudo tee /etc/eks/containerd/containerd-cached-pause-config.toml
+    sudo cp -v /etc/eks/containerd/containerd-cached-pause-config.toml /etc/containerd/config.toml
+
+    sudo cp -v /etc/eks/containerd/sandbox-image.service /etc/systemd/system/sandbox-image.service
+    sudo chown root:root /etc/systemd/system/sandbox-image.service
+    sudo systemctl daemon-reload
+    sudo systemctl start containerd
+    sudo systemctl enable containerd sandbox-image
+    /etc/eks/containerd/pull-image.sh "${PAUSE_CONTAINER}"
+    /etc/eks/containerd/pull-image.sh "${KUBE_PROXY_IMG}"
+    /etc/eks/containerd/pull-image.sh "${KUBE_PROXY_MINIMAL_IMG}"
+
+    #### Cache VPC CNI images starting with the addon default version and the latest version
+    K8S_MINOR_VERSION=$(echo "${KUBERNETES_VERSION}" | cut -d'.' -f1-2)
+    ADDON_VERSIONS=$(aws eks describe-addon-versions --addon-name vpc-cni --kubernetes-version=${K8S_MINOR_VERSION})
+    DEFAULT_VPC_CNI_VERSION=$(echo "${ADDON_VERSIONS}" | jq -r '.addons[] .addonVersions[] | select(.compatibilities[] .defaultVersion==true).addonVersion')
+    LATEST_VPC_CNI_VERSION=$(echo "${ADDON_VERSIONS}" | jq -r '.addons[] .addonVersions[] .addonVersion' | sort -V | tail -n1)
+    CNI_IMG="${ECR_URI}/amazon-k8s-cni"
+    CNI_INIT_IMG="${CNI_IMG}-init"
+    /etc/eks/containerd/pull-image.sh "${CNI_IMG}:${DEFAULT_VPC_CNI_VERSION}"
+    /etc/eks/containerd/pull-image.sh "${CNI_INIT_IMG}:${DEFAULT_VPC_CNI_VERSION}"
+    /etc/eks/containerd/pull-image.sh "${CNI_IMG}:${LATEST_VPC_CNI_VERSION}"
+    /etc/eks/containerd/pull-image.sh "${CNI_INIT_IMG}:${LATEST_VPC_CNI_VERSION}"
+    CNI_IMGS=("${CNI_IMG}:${DEFAULT_VPC_CNI_VERSION}" "${CNI_INIT_IMG}:${DEFAULT_VPC_CNI_VERSION}")
+    CNI_IMGS+=("${CNI_IMG}:${LATEST_VPC_CNI_VERSION}" "${CNI_INIT_IMG}:${LATEST_VPC_CNI_VERSION}")
+
+    CACHED_IMGS=("${PAUSE_CONTAINER}" "${KUBE_PROXY_IMG}" "${KUBE_PROXY_MINIMAL_IMG}")
+    CACHED_IMGS+=(${CNI_IMGS[@]})
+
+    #### Tag the pulled down image for all other regions in the partition
+    for region in $(aws ec2 describe-regions --all-regions | jq -r '.Regions[] .RegionName'); do
```

Comment:
I assume `AWS_DEFAULT_REGION` determines the partition?

Reply:
yep